appeals/web/src/client/components/show-more/show-more.js

```diff
@@ -72,16 +72,34 @@ const initComponentInstance = (componentInstance) => {
 	const previewText = getPreviewFromFullText(componentInstance.elements.root.innerText);
 	const labelText = componentInstance.elements.root.getAttribute(ATTRIBUTES.label);
 
-	const html = `
-		<span class="${CLASSES.text}">${previewText}</span>
-		<span class="${CLASSES.ellipsis}" aria-hidden="true">&hellip;</span>
-		<button class="${CLASSES.toggleButton}" aria-expanded="false">
-		<span class="${CLASSES.toggleButtonLabel}">${DEFAULT_OPTIONS.toggleButtonTextCollapsed}</span>
-		<span class="govuk-visually-hidden">, ${labelText}</span>
-		</button>
-	`;
+	const textSpan = document.createElement('span');
+	textSpan.className = CLASSES.text;
+	textSpan.textContent = previewText;
 
-	componentInstance.elements.root.innerHTML = html;
+	const ellipsisSpan = document.createElement('span');
```

Comment:
It's a shame this is necessary just to silence a warning, it seems quite a bit more verbose than the original. But it looks like it should all work the same. I just wanted to check whether the "DOM text reinterpreted as HTML" warning wasn't a false positive in this case. Does it actually represent a problem given the context? If so then we obviously have to address it.

Reply:
It is debatable and I agree that it is a shame as origional code seemed sound.